src/Service/Storage/Ef/DbTenantContext.cs

```diff
@@ -48,37 +48,41 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
 
         modelBuilder.Entity<EFStoredCredential>(b =>
         {
-            b.HasQueryFilter(c => c.Tenant == Tenant);
             b.HasKey(x => new { x.Tenant, x.DescriptorId });
             b.Property(x => x.DescriptorTransports).HasConversion(
                 v => JsonSerializer.Serialize(v, jsonOptions),
                 v => JsonSerializer.Deserialize<AuthenticatorTransport[]>(v, jsonOptions));
         });
 
         modelBuilder.Entity<TokenKey>()
-            .HasQueryFilter(c => c.Tenant == Tenant)
             .HasKey(c => new { c.Tenant, c.KeyId });
 
 
         modelBuilder.Entity<AccountMetaInformation>()
-            .HasQueryFilter(c => c.Tenant == Tenant)
             .Ignore(c => c.AdminEmails)
             .HasKey(x => x.AcountName);
 
         modelBuilder.Entity<ApiKeyDesc>(b =>
         {
-            b.HasQueryFilter(c => c.Tenant == Tenant);
             b.HasKey(x => new { x.Tenant, x.Id });
             b.Property(x => x.Scopes)
                 .HasConversion(
                     v => string.Join(',', v),
                     v => v.Split(',', StringSplitOptions.RemoveEmptyEntries));
         });
 
-        var ap = modelBuilder.Entity<AliasPointer>()
-            .HasQueryFilter(c => c.Tenant == Tenant)
+        modelBuilder.Entity<AliasPointer>()
             .HasKey(x => new { x.Tenant, x.Alias });
 
+        if (Tenant != null)

```

Comment:
@jonashendrickx I'm a bit hesitant to this change. We want this code path to be very defensive. If we make this change with`if (Tenant != null)` we should make other defensive measures in the TenantProvider to throw exception if value is null, or only allowing a null value if the ITenantProvider returns true for property `AllowNull` or `NonScoped`etc.